core/templates/dev/head/editor/feedback_tab/FeedbackTab.js

```diff
@@ -19,162 +19,256 @@
-            newThreadText: newThreadText
-          });
-        };
-
-        $scope.cancel = function() {
-          $modalInstance.dismiss('cancel');
-        };
-      }]
-    }).result.then(function(result) {
-      threadDataService.createNewThread(
-        result.newThreadSubject, result.newThreadText, function() {
-          $scope.clearActiveThread();
-        });
-    });
-  };
-
-  // TODO(Allan): Implement ability to edit suggestions before applying.
-  $scope.showSuggestionModal = function() {
-    $modal.open({
-      templateUrl: 'modals/editorViewSuggestion',
-      backdrop: true,
-      size: 'lg',
-      resolve: {
-        suggestion: function() {
-          return $scope.activeThread.suggestion;
-        }
-      },
-      controller: [
-        '$scope', '$modalInstance', 'suggestion',
-        function($scope, $modalInstance, suggestion) {
-        $scope.oldContent = suggestion.state_content.old_content;
-        $scope.newContent = suggestion.state_content.new_content;
-
-        $scope.acceptSuggestion = function() {
-          $modalInstance.close({
-            action: 'accept'
+    var _resetTmpMessageFields = function() {
+      $scope.tmpMessage.status = $scope.activeThread ?
+        $scope.activeThread.status : null;
+      $scope.tmpMessage.text = '';
+    };
+
+    $scope.clearActiveThread = function() {
+      $scope.activeThread = null;
+      _resetTmpMessageFields();
+    };
+
+    $scope.showCreateThreadModal = function() {
+      $modal.open({
+        templateUrl: 'modals/editorFeedbackCreateThread',
+        backdrop: true,
+        resolve: {},
+        controller: ['$scope', '$modalInstance', function(
+          $scope, $modalInstance) {
+          $scope.newThreadSubject = '';
+          $scope.newThreadText = '';
+
+          $scope.create = function(newThreadSubject, newThreadText) {
+            if (!newThreadSubject) {
+              warningsData.addWarning('Please specify a thread subject.');
+              return;
+            }
+            if (!newThreadText) {
+              warningsData.addWarning('Please specify a message.');
+              return;
+            }
+
+            $modalInstance.close({
+              newThreadSubject: newThreadSubject,
+              newThreadText: newThreadText
+            });
+          };
+
+          $scope.cancel = function() {
+            $modalInstance.dismiss('cancel');
+          };
+        }]
+      }).result.then(function(result) {
+        threadDataService.createNewThread(
+          result.newThreadSubject, result.newThreadText, function() {
+            $scope.clearActiveThread();
           });
-        };
+      });
+    };
+
+    var _isSuggestionOpen = function() {
+      return $scope.activeThread.status === 'open';
+    };
 
-        $scope.rejectSuggestion = function() {
-          $modalInstance.close({
-            action: 'reject'
+    var _isSuggestionValid = function() {
+      return explorationStatesService.getStates()[
```

Comment:
It would be cleaner to add a hasState() method to explorationStatesService (in editor/EditorServices.js) and call that instead, I think. The hasState() method does:

    return _states.hasOwnProperty(stateName);